app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :logged_in_user?, only: %i[new create]
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.user_id = current_user
+    if @post.save
+      flash[:success] = 'Post created successfully'
+      redirect_to root_path
+    else
+      render 'new'
+    end
+  end
+
+  def index
+    @posts = Post.all
+    @signed_in = current_user.present?
```

Comment:
Why this line?
I don't think it's needed because there is `current_user`

Reply:
changed